manager/controllers/app/fybrikapplication_controller.go

```diff
@@ -59,17 +62,34 @@ type FybrikApplicationReconciler struct {
 
 // Reconcile reconciles FybrikApplication CRD
 // It receives FybrikApplication CRD and selects the appropriate modules that will run
-// The outcome is a Plotter containing multiple Blueprints that run on different clusters
+// The outcome is a Plotter that describes the modules required, the clusters in which they should run, and the overall data flows.
 func (r *FybrikApplicationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	log := r.Log.WithValues("fybrikapplication", req.NamespacedName)
+	log := r.Log.With().Str("fybrikapplication", req.NamespacedName.String()).Logger()
+
 	// obtain FybrikApplication resource
 	applicationContext := &api.FybrikApplication{}
 	if err := r.Get(ctx, req.NamespacedName, applicationContext); err != nil {
-		log.V(0).Info("The reconciled object was not found")
+		// TODO - Should this be fatal rather than error level?
+		log.Error().Err(err).Str(logging.FYBRIKAPPUUID, "Unknown").Msg("The reconciled object was not found")
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
+
+	// Add the globally unique identifier for the FybrikApplication instance to the annotations for use in logging
+	// Only need to do this once, and not on every update to this resource
+	uuid := utils.GetFybrikApplicationUUID(applicationContext)
```

Comment:
Why is uuid added to annotations? Where are annotations used and why is this field duplicated in the resource?

Reply:
uuid will be added to annotations of fybrikapplication, plotter, blueprint and then it needs to be passed to each of the modules in the data plane.

Reply:
My point is that uuid can be added to annotations of plotter, blueprint and modules without changing FybrikApplication. 
Two technical concerns I have: 
1. Where do you update FybrikApplication itself, not the status? The only place it's done is inside reconcileFinalizers where finalizers are added for the first generation. 
2. Any update to the resource will cause an extra reconcile loop.